src/main_functions.c

```diff
@@ -0,0 +1,530 @@
+    	int success = generate_suggestions(misspelled[i], dict, suggestions);
+
+    	if(success != -1) {
+    	printf("Possible replacements for word %s are: ", misspelled[i]);
+    	printf("0: Delete Word. ");
+    	printf("1: No replacement. ");
+        int j = 0;
+    	for ( ; j < max_no_suggestions; j++) {
+    		printf("%d : %s ", j+2, suggestions[j]);
+
+    	}  	
+
+       }
+       printf("\n\n");
+
+    
+    //gets replacement choice from user
+    int number2;
+    printf("Enter the number of the replacement: ");
+
+    int check = scanf("%d", &number2);
+    assert(check >= 0);
+
+    if (number2 == 0) {
+    	printf("Deleting %s.", misspelled[i]);
+    	correct_line(line_copy, misspelled[i], "");
+    	printf("New sentence is: \n\n");
+     	printf("%s\n\n", line_copy);
+     	//printf("%s\n", badwords[i+1]);
+     	printf("%s\n", underline_misspelled_sentence(misspelled, line_copy, i+1));
+
+
+    }
+
+    else if (number2 == 1) {
+    	printf("No changes made to %s. \n\n", misspelled[i]);
+
+    }
+
+    else if (number2 > (max_no_suggestions+1)) {
+    	shell_error("Please enter a valid number");
+    }
+
+    else if (number2 != 1 || number2 != 0) { //1 if no replacement needed, 0 if word deleted
+    	printf("Replacing %s with %s \n", misspelled[i], suggestions[number2-2]);
+    	correct_line(line_copy, misspelled[i], suggestions[number2-2]); //modifies line function
+    	printf("New sentence is: \n");
+     	printf("%s\n\n", line_copy);
+     	printf("%s\n", underline_misspelled_sentence(misspelled, line_copy, i+1));
+    }
+
+
+	i++;	// added loop changer
+  }
+
+	return line_copy;
+	// need a way for string to (a) preserve punctuations and (b) 
+	// @Sarika this would be where the program needs replace_word, ignore_word, alternate_spelling
+}
+
+
+char** interactive_mode(char* filename, dict_t* dict, int* quit) //will pass in dictionary later
+{
+
+	size_t max_terminal_line_length = 500; // estiate
+	char** lines;
+	lines = parse_file(filename);
+
+	// step through phases
+	int i=0;
+	while (lines[i] != NULL) {	// potential error - one empty line in the middle of two full?	
+		int linenumber = i+1;
+		if (strlen(lines[i]) <= max_terminal_line_length) {
+			lines[i] = edit_interactive(lines[i], dict, linenumber);
+		}//edit interactive is called for each line 
+		else {
+			//deniz code here?
+
+		}
+		i++;
+	}
+
+    *quit = 0;
+	return lines;
+}
+
+/* 
+	IV. Batch Mode
+ */
+// Adapted from sarika's edit_interactive
+// returns corrections to a list that contains misspelled words, or returns a corrected char** string
+// that can represent an array
+char* edit_batch(char* line, dict_t* dict, int verbosity)
+{
+    char *line_copy = malloc(strlen(line));
+    strcpy(line_copy, line); //maintain a copy of the line to preserve original line: line will be parsed into individual words
+    int max_no_suggestions = 1; //need only one suggestion
+
+    int length = strlen(line)/3; //approximate 3 chars per word to be safe
```

Comment:
NO. DON'T DO THIS. SERIOUSLY. SEE ABOVE.

Reply:
done